profiler/profiler.go

```diff
@@ -70,6 +70,9 @@ func newProfiler(opts ...Option) (*profiler, error) {
 		opt(cfg)
 	}
 	if cfg.apiKey != "" {
+		if !apiKeyCheck(cfg.apiKey) {
+			return nil, fmt.Errorf("API key has incorrect format; try checking your DD_API_KEY environment variable if submitting an API key is necessary for your configuration")
```

Comment:
```suggestion
			return nil, errors.New("API key has incorrect format; try checking your DD_API_KEY environment variable if submitting an API key is necessary for your configuration")
```

Reply:
Good point overall, but should we hold back on printing valid API keys--would they be considered secret?  I'm thinking of the scenario where the user copy-pasted an API key with trailing whitespace, or the general case where an invalid API key contains (or is trivially derived from) a valid API key.

If so, what do you think about providing up to the first eight characters of the submitted key, right-padded with asterisks to the length (this is what they do in the Java profiler)?  Any other ideas?

Reply:
I think you're right. We shouldn't print out the API key. Good point.

Reply:
Sounds good.